Cython/Utility/ModuleSetupCode.c

```diff
@@ -1042,6 +1056,23 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_SET_SIZE(obj, size) Py_SIZE(obj) = (size)
 #endif
 
+#if CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS && PY_VERSION_HEX >= 0x030d00b1
+#define __Pyx_PyList_GetItemRef(o, i) PyList_GetItemRef(o, i)
+static CYTHON_INLINE int __Pyx_PyDict_GetItemRef(PyObject *dict, PyObject *key, PyObject **result) {
+  return PyDict_GetItemRef(dict, key, result);
+}
+#else
+#define __Pyx_PyList_GetItemRef(o, i) __Pyx_NewRef(PyList_GET_ITEM(o, i))
```

Comment:
Since we're not doing any error checking here, this should depend on `CYTHON_ASSUME_SAFE_MACROS` and actually also `!CYTHON_AVOID_BORROWED_REFS` since we're preferring a borrowed reference over a strong returned reference here.

Reply:
This looks unresolved.

Reply:
Understood. Thanks for the context!